### ISSUE TITLE
Steam Deck support

### ISSUE THREAD

**Issue by dennwc** [opennox/opennox#542]
This will be the first mobile platform that we directly target.

Nothing special about hardware (except gamepad controls). It uses Arch Linux as a base, so versions of `libc` doesn't match. OS has Flatpak support by default.

In order to support it properly, we need:
- #539 
- #477 (or a separate build for a different `libc`)
- #541
- #540

**Comment by Evengard** [comment]
Let's be honest - the majority of the problems is due to us not supporting 64 bits, which are native for most current platforms. Which is because of how code is structured. Should we even attempt to support other platforms right now without trying to address this issue? Because that would end up being a packaging nightmare. We already are kinda hitting it in the flatpack world, and it will get even worse.

**Comment by dennwc** [comment]
The current issue is not because of 64 bit, it's because of incompatible `libc` version that will happen as long as we use any C code (including OpenGL), regardless if it's 32 bit or 64 bit.

As for supporting new platforms, Deck is just a Linux PC, nothing exotic. We were asked about Flatpak support numerous times, so it might be a good excuse to implement it.

Also, I think mobile platforms in general are important for the project. Nox might be is a good fit for Deck-like devices both because of resolution and available controls. Hope people will find a bit more time to play it on a mobile platform. At least in my experience with Deck, I definitely found a few minutes here and there to play some games (both old and new).

**Comment by thematho** [comment]
I'm playing it in the Steam deck currently, and works but I didn't had time to actually play it a lot and see if there is any bug further into the game or multiplayer.

- OpenNox  [v1.8.11](https://github.com/noxworld-dev/opennox/releases/tag/v1.8.11) 
- Official game installed from GOG.com using he Heroic Launcher.
- Compatiblity configuration: **`opennox-hd.exe`** with **Proton 7.0-4** and **GE-Proton7-36**

The good thing about the Flatpak is that would be very easy to get updates. Snap is not easy to install on the Deck, (maybe it's not possible at all) I spend some time but there are a lot of missing libraries and you need to disable some security options from steam os. I couldn't make it work so I downloaded the zip build instead

About the gamepad support; Steam inputs are very powerful. i was trying to play around and make it work better but I found [some bugs](https://steamcommunity.com/app/1675200/discussions/0/3395175706738529830/) on steam input, I'll post a message with a video showcasing the controller config with menus for manual Phonemes and joystick run/walk speed, this could be an alternative

I modified the game resolution manually with the `opennox.xml` file. I would suggest that default resolution should be lower than the native, the character looks a little to small in my opinion. Can this be customised? That would be ideal, since you can also use the Deck with external monitors.

Native resolution `1280x800`
![image](https://user-images.githubusercontent.com/1429270/195048244-136d57f6-7fda-41d4-9d49-5beba6a10271.png)
![image](https://user-images.githubusercontent.com/1429270/195048728-43a28de2-7ff9-40c2-bb0b-1719704bff6e.png)

This looks ideal to me `960x600`
![image](https://user-images.githubusercontent.com/1429270/195048371-91fbe978-46ae-4007-a751-b66293db4b48.png)
![image](https://user-images.githubusercontent.com/1429270/195048847-e3ff25af-fc36-4c54-aba2-27f1f9390839.png)

**Comment by Evengard** [comment]
The libc incompatibility is mostly because the actual platform libc is 64 bits, while we require the 32 bit one. There are rare cases of glibc/uclibc/etc incompatibilities, but Arch (if Steam Deck is running on Arch) is using the good old glibc, so that's not the case here. So yeah, the libc incompatibility is just a symptom of the global problem.
Even mobile platforms are 64 bits rn, and they usually don't have any 32 bit libraries at all. So we will need to provide them somehow. Which is where the packaging hell is about to start...
And libc is just one of theese libs.

**Comment by blmarket** [comment]
Good thing is at least we can run it using Windows binary + proton...

As another Arch Linux (not steam deck though) user, I have no problem running the game in native Linux. Can someone share some details about the libc error? so that I can test it out. For example #540 I think it's more about missing library packages either https://archlinux.org/packages/multilib/x86_64/lib32-sdl2/ or https://archlinux.org/packages/extra/x86_64/sdl2/ (not sure it needs 32bit or 64bit library)

**Comment by dennwc** [comment]
Yeah, I may be wrong about `libc`, maybe I confused it with a different test I did.

**Comment by dennwc** [comment]
Native Flatpak test build is [now available](https://github.com/noxworld-dev/opennox/issues/477#issuecomment-1378462246).

**Comment by thematho** [comment]
> Native Flatpak test build is [now available](https://github.com/noxworld-dev/opennox/issues/477#issuecomment-1378462246).

Tested on Desktop Mode and game Mode. Seems to work great.

**Desktop mode:**
![open nox flatpack desktop](https://user-images.githubusercontent.com/1429270/222896070-93629e9a-3852-444e-9188-f4443ae85bdc.png)

**Game Mode:**
![opennox game mode flatpak](https://user-images.githubusercontent.com/1429270/222896113-faf4a464-3b16-45a0-815c-5277a88aaf83.jpg)
